numpyro/distributions/continuous.py

```diff
@@ -1,441 +1,373 @@
-        mu = (pA - pB) / d   # correction sign
-        mu2 = 1 + (a * pA - b * pB) / d - mu * mu
-        return mu, mu2, None, None
-
-
-class uniform_gen(jax_continuous):
-    _support_mask = constraints.unit_interval
-
-    def _rvs(self):
-        return random.uniform(self._random_state, shape=self._size)
-
-    def _cdf(self, x):
-        return x
-
-    def _ppf(self, q):
-        return q
-
-    def _stats(self):
-        return 0.5, 1.0 / 12, 0, -1.2
-
-    def _entropy(self):
-        return 0.0
-
-
-beta = beta_gen(a=0.0, b=1.0, name='beta')
-cauchy = cauchy_gen(name='cauchy')
-expon = expon_gen(a=0.0, name='expon')
-gamma = gamma_gen(a=0.0, name='gamma')
-halfcauchy = halfcauchy_gen(a=0.0, name='halfcauchy')
-halfnorm = halfnorm_gen(a=0.0, name='halfnorm')
-lognorm = lognorm_gen(a=0.0, name='lognorm')
-norm = norm_gen(name='norm')
-pareto = pareto_gen(a=1.0, name="pareto")
-t = t_gen(name='t')
-trunccauchy = trunccauchy_gen(name='trunccauchy')
-truncnorm = truncnorm_gen(name='truncnorm')
-uniform = uniform_gen(a=0.0, b=1.0, name='uniform')
+    def log_prob(self, value):
+        if self._validate_args:
+            self._validate_sample(value)
+        return -((value - self.loc) ** 2) / (2.0 * self.scale ** 2) \
+               - np.log(self.scale) - np.log(np.sqrt(2 * np.pi))
+
+    @property
+    def mean(self):
+        return np.broadcast_to(self.loc, self.batch_shape)
+
+    @property
+    def variance(self):
+        return np.broadcast_to(self.scale ** 2, self.batch_shape)
+
+
+class LogNormal(TransformedDistribution):
+    arg_constraints = {'loc': constraints.real, 'scale': constraints.positive}
+    support = constraints.positive
+    reparametrized_params = ['loc', 'scale']
+
+    def __init__(self, loc, scale, validate_args=None):
+        base_dist = Normal(loc, scale)
+        self.loc, self.scale = base_dist.loc, base_dist.scale
+        super(LogNormal, self).__init__(base_dist, ExpTransform(), validate_args=validate_args)
+
+    @property
+    def mean(self):
+        return np.exp(self.loc + self.scale ** 2 / 2)
+
+    @property
+    def variance(self):
+        return (np.exp(self.scale ** 2) - 1) * np.exp(2 * self.loc + self.scale ** 2)
+
+
+class Pareto(TransformedDistribution):
+    arg_constraints = {'alpha': constraints.positive, 'scale': constraints.positive}
+    support = constraints.real
+
+    def __init__(self, scale, alpha, validate_args=None):
+        batch_shape = lax.broadcast_shapes(np.shape(scale), np.shape(alpha))
+        self.scale, self.alpha = np.broadcast_to(scale, batch_shape), np.broadcast_to(alpha, batch_shape)
+        base_dist = Exponential(self.alpha)
+        transforms = [ExpTransform(), AffineTransform(loc=0, scale=self.scale)]
+        super(Pareto, self).__init__(base_dist, transforms, validate_args=validate_args)
+
+    @property
+    def mean(self):
+        # mean is inf for alpha <= 1
+        a = lax.div(self.alpha * self.scale, (self.alpha - 1))
```

Comment:
native division here would throw an exception when `alpha = 1`, whereas `lax.div` is safe.

Reply:
This does not happen in my system though. Maybe this is fixed in newer version of JAX.

Reply:
Hmm..this is a python thing. Note that `np.broadcast_to(1, ()) / np.broadcast_to(0, ())` would give a `ZeroDivisionError` since those values are not jax scalars but python floats.

Reply:
Nice catch! This is good to learn!